frontend/src/lib/components/ChartFilter/chartFilterLogic.ts

```diff
@@ -2,43 +2,47 @@ import { kea } from 'kea'
 import { router } from 'kea-router'
 import { objectsEqual } from 'lib/utils'
 import { chartFilterLogicType } from './chartFilterLogicType'
-import { ChartDisplayType, ViewType } from '~/types'
+import { ChartDisplayType, FunnelVizType, ViewType } from '~/types'
 
 export const chartFilterLogic = kea<chartFilterLogicType>({
     actions: () => ({
-        setChartFilter: (filter: ChartDisplayType) => ({ filter }),
+        setChartFilter: (filter: ChartDisplayType | FunnelVizType) => ({ filter }),
     }),
     reducers: {
         chartFilter: [
-            null as null | ChartDisplayType,
+            null as null | ChartDisplayType | FunnelVizType,
             {
                 setChartFilter: (_, { filter }) => filter,
             },
         ],
     },
     listeners: ({ values }) => ({
-        setChartFilter: () => {
-            const { display, ...searchParams } = router.values.searchParams // eslint-disable-line
+        setChartFilter: ({ filter }) => {
+            const { display, funnel_viz_type, ...searchParams } = router.values.searchParams // eslint-disable-line
             const { pathname } = router.values.location
-            searchParams.display = values.chartFilter
-
-            if (!objectsEqual(display, values.chartFilter)) {
+            const isFunnelVizType = filter === 'steps' || filter === 'time_to_convert' || filter === 'trends'
+            if (isFunnelVizType) {
+                searchParams.funnel_viz_type = filter
+                searchParams.display = ChartDisplayType.FunnelViz
+            } else {
+                searchParams.display = values.chartFilter
+            }
```

Comment:
Won't hold up the PR for it, but this is a good example of a random list of constants somewhere that you will be forgotten the next time your colleague adds a new `FunnelVizType` and doesn't know of this line.

This is also a good usecase for [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates).

Reply:
oh yeah! I tried to use `Object.keys(FunnelVizType).includes(filter)` while I was debugging something and thought it was a cause of the issue (which it turned out not to be) and forgot to clean this back up again, but I'll look to use this type predicate thing instead for it while I'm fixing merge conflicts anyway